src/D2L.CodeStyle.Analyzers/Language/LikelyArgumentMismatchAnalyzer.cs

```diff
@@ -0,0 +1,773 @@
+
+			public override int GetHashCode() {
+				unchecked {
+					int hash = 13;
+					hash = ( hash * 7 ) + this.Name.GetHashCode();
+
+					return hash;
+				}
+			}
+
+			public static bool operator == (ParameterDetails p1, ParameterDetails p2 ) {
+				return p1.Name == p2.Name;
+			}
+
+			public static bool operator !=( ParameterDetails p1, ParameterDetails p2 ) {
+				return p1.Name != p2.Name;
+			}
+		}
+
+		private static IEnumerable<Edge> MatchArguments(
+			SemanticModel model,
+			SeparatedSyntaxList<ArgumentSyntax> arguments
+		) {
+			foreach( var arg in arguments ) {
+				// Matched explicitly to a parameter
+				// Trust the developer
+				if( arg.NameColon != null ) {
+					yield break;
+				}
+
+				// Not a variable, so no name to compare
+				var identifer = arg.Expression as IdentifierNameSyntax;
+				if( identifer == null ) {
+					continue;
+				}
+
+				var argDetails = new ArgumentDetails(
+					name: GetArgName( identifer ),
+					syntax: identifer,
+					location: identifer.GetLocation()
+				);
+
+				IParameterSymbol param = arg.DetermineParameter(
+					model,
+					allowParams: false
+				);
+
+				// Couldn't get param for some reason, whatever
+				if( param == null ) {
+					continue;
+				}
+
+				// .Name documented as possibly empty, nothing to compare
+				if( param.Name == "" ) {
+					continue;
+				}
+
+				var paramDetails = new ParameterDetails(
+					name: param.Name,
+					type: param.Type
+				);
+
+				yield return new Edge(
+					model: model,
+					arg: argDetails,
+					param: paramDetails
+				);
+			}
+		}
+
+		private static string GetArgName( IdentifierNameSyntax arg ) {
+			return arg.Identifier.Text;
+		}
+
+		private static ITypeSymbol GetArgType( SemanticModel model, IdentifierNameSyntax arg ) {
+			var symbol = model.GetSymbolInfo( arg ).Symbol;
+
+			switch( symbol ) {
+				case IParameterSymbol param:
+					return param.Type;
+				case ILocalSymbol local:
+					return local.Type;
+				default:
+					throw new Exception();
+			}
+		}
+
+		// Adapted from https://en.wikipedia.org/wiki/Levenshtein_distance
+		private static int LevenshteinDistance( string x, string y ) {
+			int m = x.Length;
+			int n = y.Length;
+
+			if( m == 0 ) {
+				return n;
+			}
+
+			if( n == 0 ) {
+				return m;
+			}
+
+			int[] v0 = new int[ n + 1 ];
+			int[] v1 = new int[ n + 1 ];
+
+			for( int i = 0; i < n + 1; ++i ) {
+				v0[ i ] = i;
+			}
+
+			for( int i = 0; i < m; ++i ) {
+				v1[ 0 ] = i + 1;
+				for( int j = 0; j < n; ++j ) {
+					int deletionCost = v0[ j + 1 ] + 1;
+					int insertionCost = v1[ j ] + 1;
+					int substituionCost = v0[ j ] +
+						( x[ i ] == y[ j ] ? 0 : 1 );
+					v1[ j + 1 ] = Math.Min( deletionCost, Math.Min( insertionCost, substituionCost ) );
+				}
+
+				int[] tmp = v0;
+				v0 = v1;
+				v1 = tmp;
+			}
+
+			return v0[ n ];
+		}
+
+
+		#region Hungarian Algorithm
+		// https://github.com/vivet/HungarianAlgorithm/blob/master/HungarianAlgorithm/HungarianAlgorithm.cs

```

Comment:
Would it be easier to split this into a separate file?

Reply:
Yeah I think they're just fooling around right now :)

Reply:
Almost certainly.  I just wanted this in to see how effective it would be